components/builder-http-gateway/src/http/helpers.rs

```diff
@@ -479,3 +468,19 @@ fn do_group_promotion(
 
     route_message::<OriginPackageGroupPromote, NetOk>(req, &opgp)
 }
+
+fn is_worker(req: &mut Request) -> bool {
+    let session = req.extensions.get::<Authenticated>().unwrap();
+    let flags = FeatureFlags::from_bits(session.get_flags()).unwrap();
+
+    if flags.contains(privilege::BUILD_WORKER) {
```

Comment:
I think you could remove the if statement

Reply:
Yeah, good call